Firebase/Database/Public/FIRDatabase.h

```diff
@@ -46,6 +46,27 @@ FIR_SWIFT_NAME(Database)
 + (FIRDatabase *) database FIR_SWIFT_NAME(database());
 
 /**
+ * Gets a FirebaseDatabase instance for the specified URL.
+ *
+ * @param url The URL to the Firebase Database instance you want to access.
+ * @return A FIRDatabase instance.
+ */
++ (FIRDatabase *)databaseWithURL:(NSString *)url NS_SWIFT_NAME(database(url:));
+
+/**
+ * Gets a FirebaseDatabase instance for the specified URL, using the specified
+ * FirebaseApp.
+ *
+ * @param app The FIRApp to get a FIRDatabase for.
+ * @param url The URL to the Firebase Database instance you want to access.
+ * @return A FIRDatabase instance.
+ */
+// clang-format off
++ (FIRDatabase *)databaseForApp:(FIRApp *)app
```

Comment:
What happens in the case where someone passes in the same app but different URLs?

Reply:
They will end up using the same session identifier and both get deleted (which is intended).

@mikelehen Doesn't this mean that these mutli-resource apps will share a persistent cache?

Reply:
Looks like we are including the repo info as well:  

NSString* repoHashString = [NSString stringWithFormat:@"%@_%@", self.repoInfo.host, self.repoInfo.namespace];
NSString* persistencePrefix = [NSString stringWithFormat:@"%@/%@", self.config.sessionIdentifier, repoHashString];

Reply:
We include the namespace name as part of the persistence directory. https://github.com/firebase/firebase-ios-sdk/blob/798768380a45d81f9963d9ea51cca408f837a332/Firebase/Database/Core/FRepo.m#L120